### ISSUE TITLE
[Suggestion] Allow to customise prefix for SOAP Envelope

### ISSUE THREAD

**Issue by mubaarakhassan** [channeladam/ChannelAdam.Soap#10]
Hello, 

I have a feature suggestion. Currently we need to create a SOAP envelope with a defined namespace prefix.  When building a new SOAP envelope with your library we get the generated output which has the namespace prefix `env`. 

Is it possible to make this customisable. Currently we don't have any control over namespace prefixes or on the soap envelope tag. Looking at the code is there anyway we can insert our own NamespacePrefixConstants.SoapEnvelope? Possibly by adding a custom SoapBuilder feature like AddCustomSoapPrefix? 

Also thanks for this library. It makes it easier for building SOAP.

**Comment by channeladam** [comment]
Happy to review a design and accept a PR.

If you do want to proceed, please discuss here more details of the design you are suggesting, and the side effects - so we can agree on a design before any coding.

**Comment by mubaarakhassan** [comment]
Hello,

Thanks for being open to add new features. 

My suggestion is to extend the current usage of ```ISoap11EnvelopeBuilder``` and add a new function called ```SetPrefix(string prefix)```. Allowing to add prefix to the SoapMaker.

The detailed design would look something like this.

```csharp
namespace ChannelAdam.Soap.Abstractions
{
    using System.Xml.Linq;

    public interface ISoap11EnvelopeBuilder
    {
        ISoap11EnvelopeBuilder SetPrefix(string prefix);

        ISoap11BodyBuilder WithBody { get; }

        ISoap11HeaderBuilder WithHeader { get; }

        XContainer Build();
    }
}
```

```csharp
namespace ChannelAdam.Soap.Internal
{
    public class Soap11EnvelopeBuilder : ISoap11EnvelopeBuilder
    {
        #region Private Fields

        private string _prefix;
        private Soap11BodyBuilder? bodyBuilder;
        private Soap11HeaderBuilder? headerBuilder;
        #endregion Private Fields

        // Internal Constructors

        // region Public Properties

        #region Public Methods

        public ISoap11EnvelopeBuilder SetPrefix(string prefix)
        {
            _prefix = prefix;
            return this;
        }

        public XContainer Build()
        {
            return Soap11Maker.CreateSoapEnvelope(this.headerBuilder?.Build(), this.bodyBuilder?.Build(), _prefix);
        }

        #endregion Public Methods
    }
}
```
As you can see when Build is called the the prefix will be placed and then further used to place the namespace prefix.  Obviously the prefix will be a nullable parameter inside ```CreateSoapEnvelope``` and if the user does not give a prefix the ```NamespacePrefixConstants.SoapEnvelope``` will be used. 

The  ```CreateSoapEnvelope``` will then add this prefix further where it is needed.

Finally the usage would look something like this.

```csharp
var prefix = "soapenv";
var soapEnvelope = SoapBuilder.CreateSoap11Envelope()
                .SetPrefix(prefix)
                .Build()
                .ToString();
```
Hopefully this enough.

**Comment by mubaarakhassan** [comment]
I cannot think of a side effect right now. Probably if I would guess it would be adding the prefix everywhere inside ```SoapMaker```.

**Comment by channeladam** [comment]
Hi @mubaarakhassan,

That looks about right, with the following changes:
- Rename SetPrefix() to SetNamespacePrefix()
- Apply the same changes to the SOAP 1.2 classes

Of course, adding scenarios (for 1.1 and 1.2) to the unit tests / behaviour specifications will also be required.

Cool, happy for this to proceed.

**Comment by channeladam** [comment]
@mubaarakhassan I made some updates to the PR to not break backwards compatibility.
This has been released - version 3.1.0  - https://www.nuget.org/packages/ChannelAdam.Soap/3.1.0
Please let me know if you have any problems, and we'll close this issue after that.

**Comment by mubaarakhassan** [comment]
Yes, it's working as intended. Thank you very much for the update.
